src/network/networkd-route.c

```diff
@@ -781,6 +781,21 @@ int route_remove(Route *route) {
         return 0;
 }
 
+int route_remove_and_drop(Route *route) {
+        if (!route)
+                return 0;
+
+        route_cancel_request(route, NULL);
+
+        if (route_exists(route))
+                return route_remove(route);
+
+        if (route->state == 0)
```

Comment:
i wonder if one day we should add a static inline helper ROUTE_IS_OFFLINE() or so that does this check. it's not obvious what this check here does from just looking at the code.